digits/model/tasks/caffe_train.py

```diff
@@ -1184,6 +1184,66 @@ def get_layer_visualizations(self, net, layers='all'):
+                                visualizations.append(
+                                        {
+                                            'name': str(bottom),
+                                            'type': 'Activations',
+                                            'mean': mean,
+                                            'stddev': std,
+                                            'histogram': hist,
+                                            'image_html': utils.image.embed_image_html(vis),
+                                        }
+                                    )
+                                added_activations.append(bottom)
+                        if layer.name in net.params:
+                            data = net.params[layer.name][0].data
+                            if layer.type not in ['InnerProduct']:
+                                vis = self.get_layer_visualization(data)
+                            else:
+                                vis = None
+                            mean, std, hist = self.get_layer_statistics(data)
+                            visualizations.append(
+                                    {
+                                        'name': str(layer.name),
+                                        'type': 'Weights (%s layer)' % layer.type,
+                                        'mean': mean,
+                                        'stddev': std,
+                                        'histogram': hist,
+                                        'image_html': utils.image.embed_image_html(vis),
+                                    }
+                                )
+                        for top in layer.top:
+                            if top in net.blobs and top not in added_activations:
+                                data = net.blobs[top].data[0]
+                                normalize = True
+                                # don't normalize softmax layers
+                                if layer.type == 'Softmax':
+                                    normalize = False
+                                vis = self.get_layer_visualization(data, normalize=normalize)
+                                mean, std, hist = self.get_layer_statistics(data)
+                                visualizations.append(
+                                        {
+                                            'name': str(top),
+                                            'type': 'Activation',
+                                            'mean': mean,
+                                            'stddev': std,
+                                            'histogram': hist,
+                                            'image_html': utils.image.embed_image_html(vis),
+                                        }
+                                    )
+                                added_activations.append(top)                
```

Comment:
Yes, will remove it, that belongs to feature extraction code snippet, this will be part of a separate pull request .